testing/dart/channel_buffers_test.dart

```diff
@@ -30,6 +32,32 @@ void main() {
     });
   });
 
+  test('drain is async', () async {
+    const String channel = 'foo';
+    final ByteData data = _makeByteData('message');
+    final ui.ChannelBuffers buffers = ui.ChannelBuffers();
+    final ui.PlatformMessageResponseCallback callback = (ByteData responseData) {};
+    buffers.push(channel, data, callback);
+    final List<String> log = <String>[];
+    final Completer<void> completer = Completer<void>();
+    scheduleMicrotask(() { log.add('before drain, microtask'); });
+    log.add('before drain');
+    buffers.drain(channel, (ByteData drainedData, ui.PlatformMessageResponseCallback drainedCallback) async {
+      log.add('callback');
+      completer.complete();
+    });
+    log.add('after drain, before await');
+    await completer.future;
+    log.add('after await');
+    expect(log, <String>[
+      'before drain',
+      'callback',
+      'after drain, before await',
+      'before drain, microtask',
+      'after await',
+    ]);
+  });
```

Comment:
actual output is:

              'before drain',
              'after drain, before await',
              'before drain, microtask',
              'callback',
              'after await'

Reply:
ok i just forgot to recompile my code, never mind...